src/main/java/christmas/domain/discount/EventBadge.java

```diff
@@ -0,0 +1,63 @@
+package christmas.domain.discount;
+
+import static christmas.static_class.CommonStatic.NONE_NAME;
+import static christmas.static_class.DiscountStatic.SANTA_NAME;
+import static christmas.static_class.DiscountStatic.SANTA_PRICE;
+import static christmas.static_class.DiscountStatic.STAR_NAME;
+import static christmas.static_class.DiscountStatic.STAR_PRICE;
+import static christmas.static_class.DiscountStatic.TREE_NAME;
+import static christmas.static_class.DiscountStatic.TREE_PRICE;
+
+import christmas.domain.menu.component.Name;
+import christmas.domain.menu.component.Price;
+
+public enum EventBadge {
+
+    NONE(new Name(NONE_NAME), new Price(0)),
+    STAR(new Name(STAR_NAME), new Price(STAR_PRICE)),
+    TREE(new Name(TREE_NAME), new Price(TREE_PRICE)),
+    SANTA(new Name(SANTA_NAME), new Price(SANTA_PRICE));
+    private final Name badgeName;
+    private final Price totalDiscountPrice;
+
+    EventBadge(final Name badgeName, final Price totalDiscountPrice) {
+        this.badgeName = badgeName;
+        this.totalDiscountPrice = totalDiscountPrice;
+    }
+
+    public static EventBadge findByPrice(final Price price) {
+        if (price.value() > getSantaCriteria().value()) {
+            return SANTA;
+        }
+
+        if (price.value() > getTreeCriteria().value()) {
+            return TREE;
+        }
+
+        if (price.value() > getStarCriteria().value()) {
+            return STAR;
+        }
+
+        return NONE;
```

Comment:
`enum`의 순서를 적절히 지정해 두면 조건문 분기 대신 `Arrays.stream(values())`를 사용해 탐색 로직을 리팩토링 할 수 있습니다.
```suggestion
        return Arrays.stream(values())
                .filter(badge -> price.value() > badge.totalDiscountPrice.value())
                .findFirst()
                .orElse(NONE);
```

위 방법의 장점은 새로운 뱃지 enum이 추가되는 상황에서도 탐색 로직의 수정 없이 그냥 enum 상수만 추가해 주면 된다는 것에 있습니다.

enum의 순서에 의존적인 로직이 마음에 들지 않는다면 적용 가격의 최소, 최대값을 함께 저장하여 입력 값에 대해 무조건 하나의 결과만 필터링 되도록 사용할 수도 있어요.

Reply:
이것 또한 배워갑니다.. Stream을 이렇게도 활용할 수 있네요..!